app/models/queries/principals/orders/name_order.rb

```diff
@@ -35,15 +35,22 @@ def self.key
     :name
   end
 
-  private
+  ##
+  # Returns the selected user format as an array
+  # of SQL strings, i.e. "NULLIF(firstname, '') ASC NULLS LAST"
+  def self.name_order_statements(desc = false)
+    User::USER_FORMATS_STRUCTURE[Setting.user_format]
+        .map do |format|
+      attr = "#{Principal.table_name}.#{format}"
+      direction = desc ? "DESC" : "ASC"
 
-  def order
-    ordered = self.model.order_by_name
-
-    if direction == :desc
-      ordered = ordered.reverse_order
+      Arel.sql("NULLIF(#{attr}, '') #{direction} NULLS LAST")
     end
+  end
 
-    ordered
```

Comment:
This leads to the groups and placeholder users always being sorted to the end. 

An alternative would be to sort by the combined attributes:

```
Arel.sql("#{User::USER_FORMATS_STRUCTURE[Setting.user_format].join(" || ' ' || ")} #{direction = :desc ? "DESC" : "ASC"}")
```

Don't know about the performance impact, though. Maybe we need a couple of indices in the long run but I'd be willing to try it out without them for now.

Reply:
I did something similar to the user name filter now and used `concat_ws`. I'm quite happy with the solution now

Reply:
Works splendidly, thanks.